src/software/ai/hl/stp/tactic/pass_defender/pass_defender_fsm.h

```diff
@@ -82,17 +103,38 @@ struct PassDefenderFSM
         DEFINE_SML_ACTION(blockPass)
         DEFINE_SML_ACTION(interceptBall)
 
+        DEFINE_SML_STATE(DribbleFSM)
+        DEFINE_SML_GUARD(ballNearbyWithoutThreat)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(prepareGetPossession, DribbleFSM)
+
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
             *BlockPassState_S + Update_E[passStarted_G] / interceptBall_A =
                 InterceptBallState_S,
             BlockPassState_S + Update_E / blockPass_A,
             InterceptBallState_S + Update_E[ballDeflected_G] / blockPass_A =
                 BlockPassState_S,
+            InterceptBallState_S + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+            DribbleFSM_S,
+            DribbleFSM_S + Update_E[!ballNearbyWithoutThreat_G] / blockPass_A =
+                    BlockPassState_S,
             InterceptBallState_S + Update_E / interceptBall_A,
             X + Update_E / SET_STOP_PRIMITIVE_ACTION = X);
     }
 
    private:
+    /** Max ratio between distances (crease and ball) / (crease and nearest enemy) for
+     * crease to chase ball. Scale from (0, 1)
+     * |------------------------------------------|
+     * | Crease | <-----------------------> Enemy |
+     * |        | <----> Ball                 |   |
+     * |       ()         x                  ()   |
+     * |------------------------------------------|
+     */
```

Comment:
looks like this diagram is for the crease defender. you need to update it for pass defenders